src/include/catalog/postgres/pg_attribute_impl.h

```diff
@@ -0,0 +1,120 @@
+#pragma once
+
+#include <vector>
+
+#include "common/managed_pointer.h"
+#include "storage/projected_row.h"
+#include "storage/storage_defs.h"
+
+namespace noisepage::catalog {
+class DatabaseCatalog;
+}  // namespace noisepage::catalog
+
+namespace noisepage::execution::functions {
+class FunctionContext;
+}  // namespace noisepage::execution::functions
+
+namespace noisepage::storage {
+class RecoveryManager;
+class SqlTable;
+
+namespace index {
+class Index;
+}  // namespace index
+}  // namespace noisepage::storage
+
+namespace noisepage::transaction {
+class TransactionContext;
+}  // namespace noisepage::transaction
+
+namespace noisepage::catalog::postgres {
+class Builder;
+
+/** The NoisePage version of pg_attribute. */
+class PgAttributeImpl {
+ public:
+  /**
+   * Prepare to create pg_attribute.
+   * Does NOT create anything until the relevant bootstrap functions are called.
+   *
+   * @param db_oid          The OID of the database that pg_attribute should be created in.
+   */
+  explicit PgAttributeImpl(db_oid_t db_oid);
+
+  /** Bootstrap the projected row initializers for pg_attribute. */
+  void BootstrapPRIs();
+
+  /**
+   * Bootstrap:
+   *    pg_attribute
+   *    pg_attribute_oid_index
+   *    pg_attribute_name_index
+   *
+   * @param dbc             The catalog object to bootstrap in.
+   * @param txn             The transaction to bootstrap in.
+   */
+  void Bootstrap(common::ManagedPointer<DatabaseCatalog> dbc,
+                 common::ManagedPointer<transaction::TransactionContext> txn);
```

Comment:
I'm not sure this one separates out very cleanly and may need to be apart of the "core" catalog implementation that still gets handled as a unit (along with `pg_class`, `pg_index`, and `pg_namespace`) since you cannot fully define any table or index (to include these tables and their associated indexes) without all four of these tables.